eclipse/che-theia
# Use a dedicated server as webview domain

Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>



### What does this PR do?
Uses a dedicated endpoint to deploy all webviews separately from Che Theia editor endpoint.
It increases security since all webviews don't have access to the data of Che Theia window.

**Note**, this PR depends on a couple of other PRs (already merged):
- plugin unique endpoint feature introduced by the PR https://github.com/eclipse/che/pull/15432
- PR https://github.com/eclipse/che-plugin-registry/pull/335 that enables additional endpoint for the webviews

To test it:
- run Che with enabled TLS and using self-signed certificates. For the details, see the comments https://github.com/eclipse/che/issues/14035#issuecomment-519822676 and https://github.com/eclipse/che/issues/14035#issuecomment-562141679. Use Che Server from the image `quay.io/lkrejci/che-server:7.7.0-issue-15283` which includes unique endpoint feature
- run a workspace from the following Devfile:
```yaml
metadata:
  name: webviews
apiVersion: 1.0.0
components:
  - reference: https://raw.githubusercontent.com/azatsarynnyy/che-plugin-registry/master/v3/plugins/eclipse/che-theia/next/meta.yaml
    type: cheEditor

```
- see that theia sidecar has `webviews` endpoint which is used for loading webview of Welcome page

### What issues does this PR fix or reference?
closes https://github.com/eclipse/che/issues/15616





#### Release Notes



#### Docs PR
